src/main/java/de/tum/in/www1/artemis/service/iris/IrisWebsocketService.java

```diff
@@ -35,4 +42,73 @@ public void sendMessage(IrisMessage irisMessage) {
         websocketMessagingService.sendMessageToUser(userLogin, irisWebsocketTopic, irisMessage);
     }
 
+    /**
+     * Sends an exception over the websocket to a specific user
+     *
+     * @param irisSession to which the exception belongs
+     * @param throwable   that should be send over the websocket
+     */
+    public void sendException(IrisSession irisSession, Throwable throwable) {
+        if (!(irisSession instanceof IrisChatSession)) {
+            throw new UnsupportedOperationException("Only IrisChatSession is supported");
+        }
+        Long irisSessionId = irisSession.getId();
+        String userLogin = ((IrisChatSession) irisSession).getUser().getLogin();
+        String irisWebsocketTopic = String.format("%s/sessions/%d", IRIS_WEBSOCKET_TOPIC_PREFIX, irisSessionId);
+        websocketMessagingService.sendMessageToUser(userLogin, irisWebsocketTopic, new IrisWebsocketDTO(throwable));
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public static class IrisWebsocketDTO {
+
+        private final IrisWebsocketMessageType type;
+
+        private final IrisMessage message;
+
+        private final String errorMessage;
```

Comment:
What is the intention to have errorMessage here as well? To make the client application aware of unforeseen exceptions?

Reply:
The errorMessage is the raw message or the Java Exception. In case the exception that occurred is not an IrisException, this is the only field that is defined.